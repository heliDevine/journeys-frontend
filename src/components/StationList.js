import React, { useState } from 'react';
import { StationPopup } from './StationPopup';

const StationList = ({ stations }) => {
  const [selectedStation, setSelectedStation] = useState(null);
  const [showPopup, setShowPopup] = useState(false);

  const handleStationSelect = station => {
    setSelectedStation(station);
    setShowPopup(true);
  };

  const handleClosePopup = () => {
    setSelectedStation(null);
    setShowPopup(false);
  };

  return (
    <div className="station_grid_container">
      {stations.map(station => (
        <div className="station-summary" key={station.id}>
          <div className="station-text">
            Station name is {station.stationNameEN}
            <button
              className="info-button"
              type="button"
              onClick={() => handleStationSelect(station)}
            >
              More Info
            </button>
          </div>
        </div>
      ))}
      {showPopup && selectedStation && (
        <StationPopup
          selectedStation={selectedStation}
          onClose={handleClosePopup}
        />
      )}
    </div>
  );
};

export default StationList;
