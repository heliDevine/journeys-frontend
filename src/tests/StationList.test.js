import { render } from '@testing-library/react';
import StationList from '.././components/StationList';

describe('Stations', () => {
  const validProps = {
    stations: [
      { stationNameEN: 'Test station 1' },
      { stationNameEN: 'Test station 2' },
      { stationNameEN: 'Test station 3' },
    ],
  };
  it('renders correctly', () => {
    const { asFragment } = render(
      <StationList stations={validProps.stations} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
