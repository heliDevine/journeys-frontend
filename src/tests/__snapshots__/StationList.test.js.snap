// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stations renders correctly 1`] = `
<DocumentFragment>
  <div
    class="station_grid_container"
  >
    <div
      class="station-summary"
    >
      <div
        class="station-text"
      >
        Station name is Test station 1
        <button
          class="info-button"
          type="button"
        >
          More Info
        </button>
      </div>
    </div>
    <div
      class="station-summary"
    >
      <div
        class="station-text"
      >
        Station name is Test station 2
        <button
          class="info-button"
          type="button"
        >
          More Info
        </button>
      </div>
    </div>
    <div
      class="station-summary"
    >
      <div
        class="station-text"
      >
        Station name is Test station 3
        <button
          class="info-button"
          type="button"
        >
          More Info
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
